### PR TITLE
CSV handling added

### PR DESCRIPTION
**Pull request by erandacr**

### CODE REVIEW

File: components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/CSVInputReader.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
+package org.wso2.carbon.mediator.datamapper.engine.input.readers;
```
Comment (maheeka) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/CSVInputReader.java]: Add license headers

File: components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/CSVInputReader.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    public Schema getInputSchema() {
+        return inputSchema;
+    }
+
+    private void writeFieldElement(String fieldName, String valueString, String fieldType)
+            throws IOException, JSException, SchemaException, ReaderException {
+        switch (fieldType) {
+        case STRING_ELEMENT_TYPE:
```
Comment (maheeka) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/CSVInputReader.java]: formatting

Comment (erandacr) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/CSVInputReader.java]: Fixed with https://github.com/wso2/carbon-mediation/pull/748

File: components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/CSVInputReader.java (hunk 1 above)
Comment (erandacr) [inline components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/CSVInputReader.java]: Fixed with https://github.com/wso2/carbon-mediation/pull/748

**Pull request merged**